ProteanSoftware/DevTest
# Byron

Please accept my pull request for review:
The tasks were not difficult but, 
as I use a mac and local DB is not supported...
I had no time to time to switch to SQLite as a DB, so all the code is written in good faith.
The UI BDD tests In Cypress!! are passing and the single functional test is written in Jest

DID:
- completed tasks to an acceptable level given the time available
- updated to latest Angular
- introduced Jest into the project and Angular Testing Library: npm test
- introduced Cypress for confidence in tests and UI: npm run e2e to open cypress and debug if need be

TODO:
- write tests for .net and UI and achieve acceptable coverage
- fully test e2e using cypress
- use reactive forms everywhere
- create lauch.json and task files to start the ui and .net project
- work so that all services are using observables and use async pipe and push detection strategy
- have the configuration of the ui coming from an endpoint
- do not send arrays from the .net services but instead, send response models with support for paging
- refactor to reviews
- use git hooks and Husky to verify check-ins
- introduce renovate so that we are never out of date in packages and security audit
... more
